tests/test_deformers.py

```diff
@@ -768,6 +862,186 @@ def test_base_transformer():
 
     six.next(D.transform(jam_fixture))
 
+
+
+"""Deformer: Filtering"""
+# Helper function 
+
+def __test_tonic_filter(ann_orig, ann_new, cutoff):
+# raise error if original note now out of range is still included in annotation 
+#transform original, compare with new
+    j = 0
+    for i,obs1 in enumerate(ann_orig):
+        if j >= len(ann_new.data):
+            break
+        v_orig = midi2freq(pstrip(obs1.value['tonic']))#convert tonic to midi
+        if cutoff[0] < v_orig and cutoff[1] > v_orig:
+            #in range, see if annotation is kept the same
+            v_new = midi2freq(pstrip(ann_new.data[j].value["tonic"]))
+            ap_(v_orig, v_new)
+            j+=1
+
+
+
+def __test_contour_filter(ann_orig, ann_new, cutoff):
+
+    for obs1,obs2 in zip(ann_orig,ann_new):
```

Comment:
You probably need to check that ann_orig and ann_new have the same lengths.  Zip will automatically truncate to whichever of the two inputs is shorter.

Reply:
i did in __test_pitch_filter function